data_process.hpp

```diff
@@ -12,12 +12,43 @@ class data_process
   static constexpr uint32_t crc_mask_result = (1 << crc_size) - 1;
   static constexpr uint32_t crc_init_value  =  0;
 
+  static constexpr uint32_t whitening_init  =  0x5A;
+
   uint32_t crc;
 
+  class whitening_register
+  {
+    uint8_t as_unsigned;
+
+    public:
+
+    uint8_t get_bit(uint8_t x)
+    {
+      return as_unsigned & (1 << x);
+    }
+
+    void cycle()
+    {
+      as_unsigned <<= 1;
+
+      uint8_t msb_bit = as_unsigned >> 7;
```

Comment:
`const` ?

Reply:
yup, it should be const, will  be changed